wsd/DocumentBroker.cpp

```diff
@@ -1058,14 +1057,15 @@ void DocumentBroker::uploadToStorageInternal(const std::string& sessionId, bool
 
     const std::string uriAnonym = LOOLWSD::anonymizeUrl(uri);
 
-    // If the file timestamp hasn't changed, skip uploading.
+    // If the file timestamp hasn't changed, skip saving.
     const std::chrono::system_clock::time_point newFileModifiedTime
         = FileUtil::Stat(_storage->getRootFilePath()).modifiedTimepoint();
-    if (!isSaveAs && newFileModifiedTime == _lastFileModifiedTime && !isRename && !force)
+    if (!isSaveAs && newFileModifiedTime == _saveManager.getLastFileModifiedTime() && !isRename
```

Comment:
Are you sure the change to "saving" is correct?  Like, if I get the split between the Upload manager and Save manager - the Save manager should track the core state; while the Upload manager the state of the upload.

So how the timestamp of the _save_ could affect the skipping of the _save_ operation?  Conceptually, I even wonder why the Save manger should have any kind of remembering the timestamp (?)  Isn't that responsibility of the Upload?

Reply:
Right, it must be uploading. Fixed.

This is the decision to upload, hence the comparison of the timestamp of the file.

Save manager remembers the the last timestamp of the file on disk. Storage manager remembers the last timestamp of the file we uploaded. We need to track both separately to detect changes on disk (after saving in Core) and changes in Storage (for example, to detect conflicts when Storage changes the files behind our back).